### DISCUSSION TITLE
How to use Finetuned model via Standalone & API ?

### DISCUSSION THREAD

**Discussion by naikov** [erew123/alltalk_tts#259] [Q&A]
Hello guys, hope you are doing well!
So, I successfully finetuned my model and got the folder with it, which is named `canguru`, inside the folder models. So I have two folders inside `models`: default `xttsv2_2.0.2` and my `canguru` folder.
![bandicam 2024-06-28 17-36-31-539](https://github.com/erew123/alltalk_tts/assets/80168515/735e5de0-f03b-49c1-beca-8e2560b17431)
**My question is**: How can I use this finetuned model in the standalone app, and via API?
Currently, I find it a bit challenging, so any help will be useful

**Comment by erew123** [comment]
Hi @naikov 

I see you are using it within Text-gen-webui. I am also assuming you are using v1 and not the new v2 BETA. 

On v1, the easiest way to have access to that model in the TGWUI interface will be simply to rename the folder "canguru" to "trainedmodel" and when you next restart TGWUI you will see it shown in the TGWUI interface as a model you can select/switch to.

On the new V2 BETA version, there would be no need to rename the folder or make any other changes as all models are listed in a dropdown and its easy to switch between them all.

I hope that helps give you some options.

Thanks

**Reply by naikov** [reply]
Hi @erew123 Thank you for your answer. I followed the instructions, but I'm still unclear.
Where can I select/switch the model?
I use `AllTalk TTS Generator`, via provided UI and also via API requests as well.
![bandicam 2024-06-29 11-50-19-964](https://github.com/erew123/alltalk_tts/assets/80168515/f4016214-d8a5-48aa-a916-8e792929702e)

**Reply by erew123** [reply]
Hi @naikov 

In v1.9 in TGWUI you would see an additional model loader appear in the TGWUI interface https://raw.githubusercontent.com/erew123/screenshots/main/textgensettings.jpg (the TTS Method box).

Thanks

**Reply by naikov** [reply]
@erew123 How can I get this UI(on your screenshot)? I have installed a Standalone app only.

**Reply by erew123** [reply]
Oh sorry, I assumed from your screenshot you were using it as a part of TGWUI:

![image](https://github.com/erew123/alltalk_tts/assets/35898566/5b00b762-1ecc-46f7-818d-c49d5f8bf63b)

### AllTalk v1

If you have re-named the folder to "trainedmodel" then you can call a load of that model from an API request as follows:

https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-switching-model-endpoint

Alternatively you can rename your  "canguru" to the "xttsv2_2.0.2" folder name and use it as the model that way. 

Finally, if you look at the bottom of the documentation page (aka this page):

![image](https://github.com/erew123/alltalk_tts/assets/35898566/943fe41d-67af-4714-b756-608e724bdbf1)

There is an explanation of the configuration files and the modeldownload.json file, where you can specifically change the name of the model to load in.

### AllTalk v2
On version 2 you can simply put your models in the correct models folder and use the "load model" button as shown here in the first screenshot https://github.com/erew123/alltalk_tts/discussions/237

Version 2 is all linked here https://github.com/erew123/alltalk_tts/discussions/245

Thanks
